src/codegate/db/connection.py

```diff
@@ -431,7 +460,7 @@ async def get_workspaces(self) -> List[WorkspaceActive]:
         workspaces = await self._execute_select_pydantic_model(WorkspaceActive, sql)
         return workspaces
 
-    async def get_workspace_by_name(self, name: str) -> Optional[Workspace]:
+    async def get_non_deleted_workspace_by_name(self, name: str) -> Optional[Workspace]:
```

Comment:
most of the time we'd be dealing with non deleted workspaces. I would suggest keeping the name `get_workspace_by_name` since it adheres to the expectation.